### PR TITLE
chore: merge develop into 10.0-release (w/ electron/node updates)

### PR DESCRIPTION
**Pull request by tbiethman**
This is a develop into 10.0-release merge as of 5/12/22. Included here are some type tweaks and the dependency bump to Electron 18 + node 16.13.2. We'll also start building with FF98 in CI (up from FF93).

### User facing changelog

N/A

### Additional details

After this is merged, I'll follow up with a PR fixing https://github.com/cypress-io/cypress/issues/21236.

### PR Tasks

- [x] Have tests been added/updated?
- [n/a] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [n/a] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: npm/webpack-dev-server/src/patches.d.ts [hunk 1]
```diff
@@ -1,6 +0,0 @@
-/**
```
Comment (tbiethman) [inline npm/webpack-dev-server/src/patches.d.ts:1]: This is no longer necessary (and causes a type conflict if present) now that electron 18 is resolving to @types/node 16

**Comment by tbiethman** [comment]
The `test-binary-against-recipes` job is failing due to a bug fix that went into Chromium 97. The recipe no longer works as expected now that we're on Electron 18 (Chromium 100). Logged a ticket here: https://github.com/cypress-io/cypress-example-recipes/issues/772

We do not run that specific test with the `test-binary-against-recipes-chrome` job, which is why that job passes.

**Review by flotwig — APPROVED** [review]

**Review by mjhenkes — APPROVED** [review]

**Review by AtofStryker — COMMENTED** [review]
I ran a `yarn` locally and noticed a lot of deleted `__babel_fixtures__` inside the `config` package. I don't think this has anything to do with this PR and think some subsequent PR was missing some checked in files, but wanted to make sure.

File: circle.yml [hunk 2]
```diff
@@ -950,7 +955,9 @@ commands:
             yarn binary-build --platform $PLATFORM --version $(node ./scripts/get-next-version.js)
       - run:
           name: Zip the binary
-          command: yarn binary-zip --platform $PLATFORM
+          command: |
+            apt-get update && apt-get install -y zip
```
[the remark this answers is not in the document]
Comment (tbiethman) [reply circle.yml:959]: D'oh, very smooth. I'll update this.

**Comment by tbiethman** [comment]
@AtofStryker I see that as well, and in 10.0-release. Seems related to https://github.com/cypress-io/cypress/pull/21432.

**Comment by lmiller1990** [comment]
I added those fixtures in https://github.com/cypress-io/cypress/pull/21432. I don't know why it's showing as deleted locally. I will try reproduce.

**Comment by lmiller1990** [comment]
@tbiethman @AtofStryker please review: https://github.com/cypress-io/cypress/pull/21528

🙏

**Review by marktnoonan — APPROVED** [review]
Yarn looks good now 👍, manually checked out the app and things seem fine there. Percy's looking as expected too. The windows failures in CI look like they are currently also present on 10.0-release, I'm not sure about the darwin kitchensink tests and the linux failures, though they don't seem connected to the changes in this PR. @tbiethman Approving now and assuming those will pass when rerun.

**Comment by tbiethman** [comment]
Many thanks @marktnoonan!

**Pull request merged**